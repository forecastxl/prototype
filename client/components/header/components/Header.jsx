import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import Wrapper from './Wrapper'
import Title from './Title'
import UserNavigation from './UserNavigation'
import LoginButton from './LoginButton'

export function Header({ hasToken }) {
  return (
    <Wrapper>
      <Title>ForecastXL</Title>
      {hasToken ? <UserNavigation /> : <LoginButton />}
    </Wrapper>
  )
}

Header.propTypes = {
  hasToken: PropTypes.bool.isRequired
}

const mapStateToProps = state => ({
  hasToken: state.session.hasToken
})

export default connect(mapStateToProps, {})(Header)
