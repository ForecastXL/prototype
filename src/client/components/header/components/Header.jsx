import React from 'react'
import RaisedButton from 'material-ui/RaisedButton'
import Wrapper from './Wrapper'
import Logo from './Logo'

function Header() {
  return (
    <Wrapper>
      <Logo />
      <RaisedButton type="submit" label="Inloggen" primary />
    </Wrapper>
  )
}

export default Header
