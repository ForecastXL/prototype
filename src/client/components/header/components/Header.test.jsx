import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbHeader } from './Header'

describe('<DumbHeader />', () => {
  it('renders with login button when we are not authenticated', () => {
    const wrapper = shallow(<DumbHeader hasAuth={false} history={{}} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders with a profile widget when we are authenticated', () => {
    const wrapper = shallow(<DumbHeader hasAuth history={{}} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
