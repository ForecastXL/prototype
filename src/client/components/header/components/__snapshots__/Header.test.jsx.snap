// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbHeader /> renders with a profile widget when we are authenticated 1`] = `
<styled.header>
  <Logo />
  <Connect(withRouter(DumbProfileWidget)) />
</styled.header>
`;

exports[`<DumbHeader /> renders with login button when we are not authenticated 1`] = `
<styled.header>
  <Logo />
  <LoginButton
    onLogin={[Function]}
  />
</styled.header>
`;
